File: python/dsa.py

# def binarySearch(arr, tar):
#     left = 0
#     right = len(arr) -1
#     while(left <= right):
        
#         mid = int( left + (right - left)/2)
#         if(arr[mid]== tar):
#             return mid
#         elif arr[mid] < tar:
#             left = mid + 1
#         else:
#             right = mid - 1

#     return -1



# list = [-7, 3, 5, 9, 14, 16]
# target = 3
# result = binarySearch(list , target)

# print(result)


# s = "racecear"
# is_pal = True

# for i in range(len(s)//2):
#     if s[i] != s[-i-1]:
#         is_pal = False
#         break

# print(is_pal)


def merge(a, b):
    i = 0
    j= 0
    res= []
    while i < len(a) and j < len(b):
       if a[i] < b[j]:
          res.append(a[i])
          i+= 1
       else:
           res.append(b[j])
           j+= 1
    res.extend(a[i:])
    res.extend(b[j:])
    return res
     

   
print(merge([2, 8, 5], [3, 7, 11]))

File: javascript/dsa.js

// Binary Search
function binarySearch(arr, tar){
   let left = 0;
   let right = arr.length - 1;

   while(left <= right){
    let mid = parseInt( left + (right - left) / 2)
   
    
    if(arr[mid] == tar){
        return mid
    }else if(arr[mid] < tar){
        left = mid + 1
    }else{
        right = mid - 1
    }

    
   }

   return -1;
}

let array = [-1, 3, 4, 6, 7,9, 11, 45];
const target = 9;

const result = binarySearch(array, target);

console.log(result)

//Selection Sort
const list =[9, -2, 5, 3, 7, 11, -5];
const size = list.length;
let minIndex = 0;
let temp;

for(let i = 0; i < size - 1; i++){
    minIndex = i;
    for(let j =i; j < size; j++){
     if(list[minIndex] > list[j]) minIndex =j
    }
    temp = list[minIndex];
    list[minIndex] = list[i];
    list[i] = temp;
}


// for(let i =0; i < size; i++){
//     console.log(list[i]);
    
// }

class MyArray{
    constructor(){
        this.length = 0;
        this.data = {};
    }

    push(x){
        this.data[this.length] = x
        console.log(this.data[this.length]);
        
    }
}

const listt = new MyArray()
console.log(listt.push(7));

